src/app-layout/visual-refresh-toolbar/toolbar/drawer-triggers.tsx

```diff
@@ -136,22 +152,62 @@ export function DrawerTriggers({
+          return (
+            <TriggerButton
+              ariaLabel={item.ariaLabels?.triggerButton}
+              ariaExpanded={activeGlobalDrawersIds.includes(item.id)}
+              ariaControls={activeGlobalDrawersIds.includes(item.id) ? item.id : undefined}
+              className={clsx(
+                styles['drawers-trigger'],
+                testutilStyles['drawers-trigger'],
+                testutilStyles['drawers-trigger-global']
+              )}
+              iconName={item.trigger.iconName}
+              iconSvg={item.trigger.iconSvg}
+              key={item.id}
+              onClick={() => {
+                onActiveGlobalDrawersChange && onActiveGlobalDrawersChange(item.id);
+              }}
+              ref={globalDrawersFocusControl?.refs[item.id]?.toggle}
+              selected={activeGlobalDrawersIds.includes(item.id)}
+              badge={item.badge}
+              testId={`awsui-app-layout-trigger-${item.id}`}
+            />
+          );
+        })}
         {overflowItems.length > 0 && (
           <OverflowMenu
             items={overflowItems}
             ariaLabel={overflowMenuHasBadge ? ariaLabels?.drawersOverflowWithBadge : ariaLabels?.drawersOverflow}
-            customTriggerBuilder={({ onClick, triggerRef, ariaLabel, ariaExpanded, testUtilsClass }) => (
-              <TriggerButton
-                ref={triggerRef}
-                ariaLabel={ariaLabel}
-                ariaExpanded={ariaExpanded}
-                badge={overflowMenuHasBadge}
-                className={clsx(styles['drawers-trigger'], testutilStyles['drawers-trigger'], testUtilsClass)}
-                iconName="ellipsis"
-                onClick={onClick}
-              />
-            )}
-            onItemClick={event => onActiveDrawerChange?.(event.detail.id)}
+            customTriggerBuilder={({ onClick, triggerRef, ariaLabel, ariaExpanded, testUtilsClass }) => {
+              return (
+                <TriggerButton
+                  ref={triggerRef}
+                  ariaLabel={ariaLabel}
+                  ariaExpanded={ariaExpanded}
+                  badge={overflowMenuHasBadge}
+                  className={clsx(
+                    styles['drawers-trigger'],
+                    testutilStyles['drawers-trigger'],
+                    testutilStyles['drawers-trigger-global'],
+                    testUtilsClass
+                  )}
+                  iconName="ellipsis"
+                  onClick={onClick}
+                />
+              );
+            }}
+            onItemClick={event => {
+              const id = event.detail.id;
+              if (globalDrawers.find(drawer => drawer.id === id)) {
+                onActiveGlobalDrawersChange?.(id);
+              } else {
+                onActiveDrawerChange?.(event.detail.id);
+              }
+            }}
+            globalDrawersStartIndex={globalDrawersStartIndex - indexOfOverflowItem}
```

Comment:
What is the point of passing down this index, if you can already redistribute `items`  prop here?

Reply:
Yes, I can do that, but it would require changes to the items' data structure. When global drawers are present, the items need to be grouped, but, without global drawers, they should be passed as they are. Additionally, this logic would also need to be handled within the overflow menu, where drawers are mapped to overflow menu items. So, I realized that it would be more complicated than simply introducing an additional param

Reply:
ok, the current revision looks fine